.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,71 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+name: "CodeQL"
+
+on:
+  push:
+    branches: [4.1, master]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [4.1, master]
+  schedule:
+    - cron: '0 13 * * 3'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        # Override automatic language detection by changing the below list
+        # Supported options are ['csharp', 'cpp', 'go', 'java', 'javascript', 'python']
+        language: ['java']
```

Comment:
The projects has some JNI parts written in C. I am wondering in we can also scan C code with CodeQL.

```suggestion
        language: ['java', 'cpp' ]
```

Not sure that `c` can be used in the `language` filed. I didn't find it in the docs.

Reply:
Seems like c is not supported atm

Reply:
The docs say that both C/C++ are supported. I think we can use `cpp` to enable scans for the C code.

https://docs.github.com/en/free-pro-team@latest/github/finding-security-vulnerabilities-and-errors-in-your-code/about-code-scanning

Reply:
seems like there needs to be some more setup to make this work :/

Reply:
Yeah, it looks so. Maybe we can disable autobuild and add the following

```
- run: mvn -DskipTests install
```

Or, if it doesn't work, maybe add `if` blocks and build each JNI module separately with autogen.sh + make.

Anyway, I think we can add it later.

Reply:
let me try...

Reply:
@artem-m seems like this works ? :)

Reply:
Yeah, looks like it works! :) I don't see the results at the moment in the tab "Code scanning results". But maybe they're going to be available later (it says "This usually happens when you enable code scanning for the first time").